Data/Table/Row.idr

```diff
@@ -11,6 +11,26 @@ import public Data.Table.Row.HasRows
 
 %default total
 
+export
+ByProof : forall table.
```

Comment:
I don't see the point of the `ByProof` type, and it is overly specified for its name. Can we not just use `replace`, where needed?

Reply:
Changed the name to a less general name `HasRowsProof`. It's used 3 times, enough for me to use it as an abstraction.
Is that better?